internal/services/datadog/azurerm_datadog_monitor_sso_configuration.go

```diff
@@ -0,0 +1,172 @@
+package datadog
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/datadog/mgmt/2021-03-01/datadog"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/datadog/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/datadog/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceDatadogSingleSignOnConfigurations() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceDatadogSingleSignOnConfigurationsCreateorUpdate,
+		Read:   resourceDatadogSingleSignOnConfigurationsRead,
+		Update: resourceDatadogSingleSignOnConfigurationsCreateorUpdate,
+		Delete: resourceDatadogSingleSignOnConfigurationsDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.DatadogSingleSignOnConfigurationsID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"datadog_monitor_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validate.DatadogMonitorID,
+			},
+
+			"name": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				Default:  utils.String("default"),
+			},
+
+			"enterprise_application_id": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+			},
+
+			"singlesignon_state": {
```

Comment:
can we make this `single_sign_on_enabled` for consistency with the rest of the codebase?

Reply:
It's an enum in the backend and can have other states as well like Initial.

Reply:
can we rename this per above?

Reply:
Updated